eva_submission/eload_backlog.py

```diff
@@ -38,24 +44,59 @@ def get_species_info(self):
         """Adds species info into the config: taxonomy id and scientific name,
         and assembly accession, fasta, and report."""
         with get_metadata_conn() as conn:
-            query = f"select a.taxonomy_id, b.scientific_name, d.assembly_accession " \
+            query = f"select a.taxonomy_id, b.scientific_name " \
                     f"from project_taxonomy a " \
                     f"join taxonomy b on a.taxonomy_id=b.taxonomy_id " \
-                    f"join assembly_set c on b.taxonomy_id=c.taxonomy_id " \
-                    f"join accessioned_assembly d on c.assembly_set_id=d.assembly_set_id " \
                     f"where a.project_accession='{self.project_accession}';"
             rows = get_all_results_for_query(conn, query)
-        if len(rows) != 1:
+        if len(rows) < 1:
             raise ValueError(f'No taxonomy for {self.project_accession} found in metadata DB.')
-        tax_id, sci_name, asm_accession = rows[0]
+        elif len(rows) > 1:
+            raise ValueError(f'Multiple taxonomy for {self.project_accession} found in metadata DB.')
+        tax_id, sci_name = rows[0]
         self.eload_cfg.set('submission', 'taxonomy_id', value=tax_id)
         self.eload_cfg.set('submission', 'scientific_name', value=sci_name)
-        self.eload_cfg.set('submission', 'assembly_accession', value=asm_accession)
 
+        with get_metadata_conn() as conn:
+            query = f"select distinct b.vcf_reference_accession " \
+                    f"from project_analysis a " \
+                    f"join analysis b on a.analysis_accession=b.analysis_accession " \
+                    f"where a.project_accession='{self.project_accession}';"
+            rows = get_all_results_for_query(conn, query)
+        if len(rows) < 1:
+            raise ValueError(f'No reference accession for {self.project_accession} found in metadata DB.')
+        elif len(rows) > 1:
+            raise ValueError(f'Multiple reference accession for {self.project_accession} found in metadata DB.')
+        asm_accession, = rows[0]
+        self.eload_cfg.set('submission', 'assembly_accession', value=asm_accession)
         fasta_path, report_path = get_reference_fasta_and_report(sci_name, asm_accession)
         self.eload_cfg.set('submission', 'assembly_fasta', value=fasta_path)
         self.eload_cfg.set('submission', 'assembly_report', value=report_path)
 
+    def find_local_file(self, fn):
+        full_path = os.path.join(self._get_dir('vcf'), fn)
+        if not os.path.exists(full_path):
+            self.error(f'File not found: {full_path}')
+            raise FileNotFoundError(f'File not found: {full_path}')
+        return full_path
+
+    def find_file_on_ena(self, fn, analysis):
+        @retry(tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+        def download_file(url, dest):
```

Comment:
can we put this inner function in `eload_utils`?

Reply:
Done